### PR TITLE
Add metadata and tags to mc find

### PR DESCRIPTION
**Pull request by klauspost**
## Description

Add metadata and tag search.

In both cases a MinIO sever must be on the backend. Tags and `--versions` requires the latest release.

This is required for fixing unmarshal: https://github.com/minio/minio-go/pull/1795

## Motivation and Context

Examples
```
λ mc find -json -tags="key=v*" myminio/testbucket/
{
 "status": "success",
 "type": "",
 "lastModified": "2023-01-05T10:26:13.12+01:00",
 "size": 712001,
 "key": "myminio/testbucket/out.csv",
 "etag": "",
 "metadata": {
  "X-Amz-Server-Side-Encryption": "aws:kms",
  "content-type": "text/csv"
 },
 "tags": {
  "key": "value",
  "key2": "",
  "key3": "else"
 }
}
```

Metadata:

```
λ go build&&mc find -json -metadata="content-type=text/csv" myminio/testbucket/
{
 "status": "success",
 "type": "",
 "lastModified": "2023-01-19T17:38:43.803+01:00",
 "size": 342934,
 "key": "myminio/testbucket/go2/src/cmd/go/internal/modfetch/zip_sum_test/testdata/zip_sums.csv",
 "etag": "",
 "metadata": {
  "X-Amz-Server-Side-Encryption": "aws:kms",
  "content-type": "text/csv"
 }
}
{
 "status": "success",
 "type": "",
 "lastModified": "2023-01-19T17:41:03.011+01:00",
 "size": 342934,
 "key": "myminio/testbucket/go3/src/cmd/go/internal/modfetch/zip_sum_test/testdata/zip_sums.csv",
 "etag": "",
 "metadata": {
  "X-Amz-Server-Side-Encryption": "aws:kms",
  "content-type": "text/csv"
 }
}
```

If a value is empty `-metadata="X-Amz-Server-Side-Encryption="` it must either not exist or have an empty value.

Multiple tags or metadata fields can be queried, eg `-metadata="content-type=text/csv" -metadata="X-Amz-Server-Side-Encryption=."`. In that case all of them must exist.

Regex is Go RE2 syntax.

## How to test this PR?

See examples above.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by krisis — COMMENTED** [review]
Great stuff! PR looks good. Need to update go.mod to use latest minio-go.

**Comment by klauspost** [comment]
Merged...

**Review by harshavardhana — APPROVED** [review]

**Review by krisis — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -125,3 +125,5 @@ require (
 	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
 )
+
+replace github.com/minio/minio-go/v7 => github.com/klauspost/minio-go/v7 v7.0.0-20230330164821-4ec9eb28c34e
```
Comment (julio-lopez) [inline go.mod:129]: @klauspost is this intentional?

This is breaking `go install github.com/minio/mc@latest`

Comment (harshavardhana) [reply go.mod:129]: This is not intentional, it needs to be reverted to an actual release.

**Pull request merged**